--- data/systemd/v4l2-relayd@.service.orig	2026-06-15 12:00:00.000000000 -0500
+++ data/systemd/v4l2-relayd@.service	2026-06-15 12:00:00.000000000 -0500
@@ -14,7 +14,7 @@
 ExecCondition=/usr/bin/test -n "$HEIGHT"
 ExecCondition=/usr/bin/test -n "$FRAMERATE"
 ExecCondition=/usr/bin/test -n "${CARD_LABEL}"
-ExecStart=/bin/sh -c 'DEVICE=$(grep -l -m1 -E "^${CARD_LABEL}$" /sys/devices/virtual/video4linux/*/name | cut -d/ -f6); exec /usr/bin/v4l2-relayd -i "${VIDEOSRC}" $${SPLASHSRC:+-s "${SPLASHSRC}"} -o "appsrc name=appsrc caps=video/x-raw,format=${FORMAT},width=${WIDTH},height=${HEIGHT},framerate=${FRAMERATE} ! videoconvert ! v4l2sink name=v4l2sink device=/dev/$${DEVICE}" $EXTRA_OPTS'
+ExecStart=/bin/sh -c 'DEVICE=$(grep -l -m1 -E "^${CARD_LABEL}$" /sys/devices/virtual/video4linux/*/name | cut -d/ -f6); exec /usr/bin/v4l2-relayd -i "${VIDEOSRC}" $${SPLASHSRC:+-s "$${SPLASHSRC}"} -o "appsrc name=appsrc caps=video/x-raw,format=${FORMAT},width=${WIDTH},height=${HEIGHT},framerate=${FRAMERATE} ! videoconvert ! v4l2sink name=v4l2sink device=/dev/$${DEVICE}" $EXTRA_OPTS'
 Restart=always
 PrivateNetwork=yes
 PrivateTmp=yes
