From acfa370b92604c4e6d55f79ed283ed2b05da32de Mon Sep 17 00:00:00 2001
From: Mikko Nyman <mikko@nyman.xyz>
Date: Sat, 25 Apr 2026 17:43:18 +0300
Subject: [PATCH] focaltech_moc: add new pid:0xA97A

---
 libfprint/drivers/focaltech_moc/focaltech_moc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libfprint/drivers/focaltech_moc/focaltech_moc.c b/libfprint/drivers/focaltech_moc/focaltech_moc.c
index 8bd93e4..23dac16 100644
--- a/libfprint/drivers/focaltech_moc/focaltech_moc.c
+++ b/libfprint/drivers/focaltech_moc/focaltech_moc.c
@@ -37,6 +37,7 @@ static const FpIdEntry id_table[] = {
   { .vid = 0x2808,  .pid = 0x1579,  },
   { .vid = 0x2808,  .pid = 0x077A,  },
   { .vid = 0x2808,  .pid = 0x079A,  },
+  { .vid = 0x2808,  .pid = 0xA97A,  },
   { .vid = 0,  .pid = 0,  .driver_data = 0 },   /* terminating entry */
 };

--
2.54.0

